src/frontend/src/lib/services/load-user-profile.services.ts

```diff
@@ -0,0 +1,54 @@
+import type { UserProfile } from '$declarations/backend/backend.did';
+import { createUserProfile, getUserProfile } from '$lib/api/backend.api';
+import { i18n } from '$lib/stores/i18n.store';
+import { userProfileStore } from '$lib/stores/settings.store';
+import { toastsError } from '$lib/stores/toasts.store';
+import { UserProfileNotFoundError } from '$lib/types/errors';
+import type { Identity } from '@dfinity/agent';
+import { isNullish } from '@dfinity/utils';
+import { get } from 'svelte/store';
+
+const getUncertifiedProfile = async ({
+	identity
+}: {
+	identity: Identity;
+}): Promise<UserProfile | undefined> => {
+	try {
+		return await getUserProfile({ identity, certified: false });
+	} catch (err) {
+		if (err instanceof UserProfileNotFoundError) {
+			return undefined;
+		}
+		throw err;
+	}
+};
+
+const loadCertifiedUserProfile = async ({ identity }: { identity: Identity }): Promise<void> => {
+	try {
+		const profile = await getUserProfile({ identity, certified: true });
+		userProfileStore.set({ key: 'user-profile', value: profile });
+	} catch (err) {
+		// We ignore the error because this is a background task.
+		console.error('Failed to load certified user profile.', err);
+	}
+};
+
+export const loadUserProfile = async ({ identity }: { identity: Identity }): Promise<void> => {
+	try {
+		let profile = await getUncertifiedProfile({ identity });
+		if (isNullish(profile)) {
```

Comment:
I would appreciate if you ping prodsec about this logic - i.e. if answer to query is undefined then create.

I'm sure it's fine but, just for the state of the art as you never know.

Reply:
Done https://dfinity.slack.com/archives/C01EWN833KN/p1721824932805649

Reply:
We can go on with the PR and iterate after their review - i.e. given that we asked I think it's fine to continue with the PR. Of course we should wait on their feedback before going live but, it should not block us.

Reply:
This is not yet used, so it's not even blocking us from deploying `main`.